Easydict/Feature/Configuration/NewConfiguration.swift

```diff
@@ -0,0 +1,595 @@
+        let stringValue = "\(queryTextType)"
+        UserDefaults.standard.set(stringValue, forKey: key)
+    }
+
+    func intelligentQueryTextTypeForServiceType(_ serviceType: ServiceType) -> EZQueryTextType {
+        let key = EZConstKey.constkey("IntelligentQueryTextType", serviceType: serviceType)
+        let stringValue = UserDefaults.standard.string(forKey: key) ?? "7"
+        // Convert string to int
+        let type = (stringValue as NSString).intValue
+        return EZQueryTextType(rawValue: UInt(type))
+    }
+}
+
+// MARK: Intelligent Query Mode
+
+private extension Configuration {
+    func setIntelligentQueryMode(_ enabled: Bool, windowType: EZWindowType) {
+        let key = EZConstKey.constkey("IntelligentQueryMode", windowType: windowType)
+        let stringValue = "\(enabled)"
+        UserDefaults.standard.set(stringValue, forKey: key)
+
+        let parameters = [
+            "enabled": enabled,
+            "window_type": windowType.rawValue,
+        ] as [String: Any]
+
+        EZLog.logEvent(withName: "intelligent_query_mode", parameters: parameters)
+    }
+
+    func intelligentQueryModeForWindowType(_ windowType: EZWindowType) -> Bool {
+        let key = EZConstKey.constkey("IntelligentQueryMode", windowType: windowType)
+        let defaultValue = "0"
+        // Turn on intelligent query mode by default in mini window.
+        if windowType == .mini {
+            return true
+        }
+        return UserDefaults.standard.string(forKey: key) ?? defaultValue == "1"
+    }
+}
+
+private extension Configuration {
+    func postUpdateQuickLinkButtonNotification() {
+        let notification = Notification(name: .init("EZQuickLinkButtonUpdateNotification"), object: nil)
+        NotificationCenter.default.post(notification)
+    }
+
+    func hideMenuBarIcon(hidden: Bool) {
+        if hidden {
+            EZMenuItemManager.shared().remove()
+        } else {
+            EZMenuItemManager.shared().setup()
+        }
+    }
+
+    func updateLoginItemWithLaunchAtStartup(_ launchAtStartup: Bool) {
+        let appName = Bundle.main.object(forInfoDictionaryKey: "CFBundleExecutable") as? String
+        let appBundlePath = Bundle.main.bundlePath
+
+        let script = """
+            tell application "System Events" to get the name of every login item
+            tell application "System Events"
+                set loginItems to every login item
+                repeat with aLoginItem in loginItems
+                    if (aLoginItem's name is "\(appName ?? "")") then
+                        delete aLoginItem
+                    end if
+                end repeat
+                if \(launchAtStartup) then
+                    make login item at end with properties {path:"\(appBundlePath)", hidden:false}
+                end if
+            end tell
+        """
+
+        let exeCommand = EZScriptExecutor()
+        exeCommand.runAppleScript(script) { result, error in
+            if error != nil {
```

Comment:
Compiler says `Comparing non-optional value of type 'EZError' to 'nil' always returns true`